addons/web/static/src/views/fields/char/char_field.js

```diff
@@ -53,13 +57,50 @@ export class CharField extends Component {
             isPassword: this.props.isPassword,
         });
     }
+    get hasDynamicPlaceholder() {
+        return this.props.dynamicPlaceholder && !this.props.readonly;
+    }
 
     parse(value) {
         if (this.shouldTrim) {
             return value.trim();
         }
         return value;
     }
+
+    onBlur() {
+        this.selectionStart = this.input.el.selectionStart;
+    }
+
+    async onDynamicPlaceholderOpen() {
+        await this.dynamicPlaceholder.open({
+            validateCallback: this.onDynamicPlaceholderValidate.bind(this),
+        });
+    }
+
+    async onDynamicPlaceholderValidate(chain, defaultValue) {
+        if (chain) {
+            this.input.el.focus();
+            defaultValue = defaultValue.replace("|||", "");
+            let dynamicPlaceholder =
+                "object." +
+                chain +
+                (defaultValue && defaultValue !== "" ? ` ||| ${defaultValue}` : "");
```

Comment:
You could shorten it.
```suggestion
            const dynamicPlaceholder = `{{object.${chain}${
                defaultValue?.length ? ` ||| ${defaultValue}` : ""
            }}}`;
```